### PR TITLE
dismiss controller by swipe down; refactoring selector and closure

### PR DESCRIPTION
**Pull request by alexgolutvin**

### CODE REVIEW

File: Shari-Demo/ModalTableViewController.swift [hunk 1]
```diff
@@ -58,7 +63,7 @@ class ModalTableViewController: UITableViewController, Shari.NavigationControlle
 
     func navigationControllerDidSpreadToEntire(navigationController: UINavigationController) {
 
-        self.tableView.scrollEnabled = true
+//        self.tableView.scrollEnabled = true
```
Comment (nakajijapan) [inline Shari-Demo/ModalTableViewController.swift]: Same code as above?

File: Sources/Classes/NavigationController.swift [hunk 2]
```diff
@@ -82,23 +94,25 @@ public class NavigationController: UINavigationController {
...
                     0.6,
                     delay: 0.0,
                     usingSpringWithDamping: 0.5,
                     initialSpringVelocity: 0.1,
                     options: UIViewAnimationOptions.CurveLinear,
-                    animations: { () -> Void in
-                        
-                        ModalAnimator.transitionBackgroundView(backgroundView, location: self.originalLocation)
+                    animations: { [weak self] in
+                        guard let strongslef = self else { return }
```
Comment (nakajijapan) [inline Sources/Classes/NavigationController.swift:108]: Should I use the self?
I think I may use the self variable as it is.

File: Shari-Demo/ModalTableViewController.swift (hunk 1 above)
Comment (alexgolutvin) [inline Shari-Demo/ModalTableViewController.swift]: I uncomment this

File: Sources/Classes/NavigationController.swift (hunk 2 above)
Comment (alexgolutvin) [inline Sources/Classes/NavigationController.swift:108]: For reliability. The retain circle will not be exact

**Comment by nakajijapan** [comment]
@alexgolutvin 

Great Pull request!!!!!! Thank you!!!!

**Comment by nakajijapan** [comment]
https://github.com/nakajijapan/Shari/releases/tag/0.3.0

You can confirm at v0.3.0. Please check it! :)

**Comment by alexgolutvin** [comment]
@nakajijapan 
Thank you!:)

**Pull request merged**